tests/integration/deprecated_syscalls_tests.rs

```diff
@@ -491,3 +492,106 @@ async fn test_syscall_get_contract_address_cairo0(
     // We can only verify that the syscall is implemented and goes through.
     check_os_output_read_only_syscall(os_output, block_context);
 }
+
+#[rstest]
+// We need to use the multi_thread runtime to use task::block_in_place for sync -> async calls.
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_syscall_emit_event_cairo0(
+    #[future] initial_state_cairo0: StarknetTestState,
+    block_context: BlockContext,
+    max_fee: Fee,
+) {
+    let initial_state = initial_state_cairo0.await;
+
+    let sender_address = initial_state.cairo0_contracts.get("account_with_dummy_validate").unwrap().address;
+    let test_contract = initial_state.cairo0_contracts.get("test_contract").unwrap();
+    let contract_address = test_contract.address;
+
+    let tx_version = TransactionVersion::ZERO;
+    let keys = vec![stark_felt!(2019_u16), stark_felt!(2020_u16)];
+    let data = vec![stark_felt!(2021_u16), stark_felt!(2022_u16), stark_felt!(2023_u16)];
+    let entrypoint_args = &[
+        vec![stark_felt!(u128::try_from(keys.len()).unwrap())],
+        keys,
+        vec![stark_felt!(u128::try_from(data.len()).unwrap())],
+        data,
+    ]
+    .concat();
+
+    let mut nonce_manager = NonceManager::default();
+    let tx = test_utils::account_invoke_tx(invoke_tx_args! {
+        max_fee,
+        sender_address: sender_address,
+        calldata: create_calldata(contract_address, "test_emit_event", entrypoint_args),
+        version: tx_version,
+        nonce: nonce_manager.next(sender_address),
+    });
+
+    let txs = vec![tx];
+
+    let (_pie, _os_output) = execute_txs_and_run_os(
+        initial_state.cached_state,
+        block_context.clone(),
+        txs,
+        initial_state.cairo0_compiled_classes,
+        initial_state.cairo1_compiled_classes,
+    )
+    .await
+    .expect("OS run failed");
+}
+
+#[rstest]
+// We need to use the multi_thread runtime to use task::block_in_place for sync -> async calls.
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_syscall_send_message_to_l1_cairo0(
+    #[future] initial_state_cairo0: StarknetTestState,
+    block_context: BlockContext,
+    max_fee: Fee,
+) {
+    let initial_state = initial_state_cairo0.await;
+
+    let sender_address = initial_state.cairo0_contracts.get("account_with_dummy_validate").unwrap().address;
+    let test_contract = initial_state.cairo0_contracts.get("test_contract").unwrap();
+    let contract_address = test_contract.address;
+
+    // test constants
+    const ADDRESS: u16 = 1298;
```

Comment:
What's the point of using `u16`?

Reply:
Right, I didn't realize that someone else used `u16`. Could you use `u64` though? I'd rather avoid this type of stuff ending all over the place.